crates/lint/src/sol/mod.rs

```diff
@@ -73,11 +73,15 @@ impl SolidityLinter {
         let _ = sess.enter(|| -> Result<(), diagnostics::ErrorGuaranteed> {
             // Declare all available passes and lints
             let mut passes_and_lints = Vec::new();
-            passes_and_lints.extend(gas::create_lint_passes());
             passes_and_lints.extend(high::create_lint_passes());
             passes_and_lints.extend(med::create_lint_passes());
             passes_and_lints.extend(info::create_lint_passes());
 
+            // Do not apply gas-severity rules on tests and scripts
+            if !file.ends_with(".t.sol") && !file.ends_with(".s.sol") {
```

Comment:
I think would be better to know here if file is test or script and not rely on checking extensions (there could also be mocks in test path for example). There's a utility in `ProjectPathsConfig.is_test_or_script` we can use